### PR TITLE
b/152977807: rewrite unit test to use table driven format

### PR DESCRIPTION
**Pull request by TAOXUY**

### CODE REVIEW

File: src/go/util/marshal_test.go [hunk 1]
```diff
@@ -43,28 +40,36 @@ func TestUnmarshalBytesToPbMessage(t *testing.T) {
...
-
-	wantResp = &scpb.ReportResponse{
-		ServiceConfigId: "test-id",
+	testCases := []struct {
+		getResp   proto.Message
+		wantResp  proto.Message
+		wantError string
+	}{
+		{
+			getResp: &confpb.Service{},
```
Comment (nareddyt) [inline src/go/util/marshal_test.go:49]: Can you add `desc` variable with string description? I still don't understand why there are fields populated in the `wantResp`. Like why is `Id` set in this test case, where did `test-id` come from?

File: src/go/util/marshal_test.go [hunk 2]
```diff
@@ -43,28 +40,36 @@ func TestUnmarshalBytesToPbMessage(t *testing.T) {
...
+			wantError: "not support unmarshalling",
+		},
 	}
-	getResp = &scpb.ReportResponse{}
-	_test()
 
-	wantResp = &scpb.ReportRequest{
-		ServiceConfigId: "test-id",
+	for _, tc := range testCases {
+		_test(tc.getResp, tc.wantResp, tc.wantError)
```
Comment (nareddyt) [inline src/go/util/marshal_test.go:73]: You're passing getResp, wantResp. But func parameters are ordered wantResp, getResp

File: src/go/util/marshal_test.go [hunk 3]
```diff
@@ -26,10 +26,7 @@ import (
 )
 
 func TestUnmarshalBytesToPbMessage(t *testing.T) {
-	var wantResp, getResp proto.Message
-	wantError := ""
-
-	_test := func() {
+	_test := func(wantResp, getResp proto.Message, wantError string) {
```
Comment (nareddyt) [inline src/go/util/marshal_test.go:29]: No need for func, just move this code into the test loop.

File: src/go/util/marshal_test.go (hunk 1 above)
Comment (TAOXUY) [reply src/go/util/marshal_test.go]: It is just to differentiate the gotMessage is right. Rename it to be gotMessageHolder.

Comment (nareddyt) [reply src/go/util/marshal_test.go]: Ah i see

**Pull request merged**